ext/ftp/php_ftp.c

```diff
@@ -177,11 +218,16 @@ PHP_FUNCTION(ftp_ssl_connect)
 	/* enable ssl */
 	ftp->use_ssl = 1;
 
-	RETURN_RES(zend_register_resource(ftp, le_ftpbuf));
+	object_init_ex(return_value, php_ftp_ce);
+	ftp_object_from_zend_object(Z_OBJ_P(return_value))->ftp = ftp;
 }
 /* }}} */
 #endif
 
+#define GET_FTPBUF(ftpbuf, zftp) \
+	ftpbuf = ftp_object_from_zend_object(Z_OBJ_P(zftp))->ftp; \
+	if (!ftpbuf) { RETURN_THROWS(); }
```

Comment:
This needs to actually throw something before the RETURN_THROWS :) We should also add a test for using the FTP object after ftp_close, which should trigger this condition.

Reply:
Yeah. I caught that and just finished doing both while you were commenting. :)